scan-command/src/main/resources/scan.properties

```
@@ -0,0 +1,74 @@
+#
+# Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+#
+# WSO2 LLC. licenses this file to you under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# ------------------------
+# Scan Tool error messages
+# ------------------------
+error.invalid.number.of.arguments=\
+  Invalid number of arguments, expected one argument received {0}
+
+error.empty.package=\
+  ballerina: Package is empty. Please add at least one .bal file.
+
+error.missing.config.field=\
+  Failed to load scan tool configurations: {0} is missing
```

```suggestion
  Failed to load scan tool configurations: cannot find {0}
```

What happens if multiple are unavailable?

---

The `config` field is used for pointing to the `Scan.toml` configuration file from the `Ballerina.toml` file.

```toml
# Ballerina.toml
[scan]
config = "path/to/Scan.toml"
```

So this error was made to be shown if the user does not define the path to the configuration file as follows:

```toml
# Ballerina.toml
[scan]
# Missing config field
```

Should we consider that the  user will point to multiple configuration files?

---

Can you share what the diagnostic will look like?

---

Yes,

If we consider a Ballerina project:
```
📦ballerina-project
 ┗ 📜.devcontainer.json
 ┗ 📜.gitignore
 ┗ 📜Ballerina.toml
 ┗ 📜main.bal
 ┗ 📜Scan.toml
```

With the `Ballerina.toml` defined as:
```toml
[scan]
# configPath is missing
```

The result displayed in the console would be as follows:
```bash
PS C:\Users\Tharana Wanigaratne\Desktop\ballerina-project> bal scan

Running Scans...
Failed to load scan tool configurations: cannot find configPath
```

---

So the error message is fixed, right? Why do we need to pass configPath as a param.

Btw, use quotes with configPath. We should either refer directly by name with quotes or have it as a proper term.

---

Since it's a field to be checked in the toml file, in the Utility method [it's used as a static constant](https://github.com/ballerina-platform/static-code-analysis-tool/blob/16c40747349c5d92f3ddce4f6ba79b1b20ebd32e/scan-command/src/main/java/io/ballerina/scan/utils/Constants.java#L47), This is why it was provided as a param.

Will remove the param and have it there as a fixed message.